src/main.py

```diff
@@ -117,8 +117,11 @@ def check_call(args):
     subprocess.check_call(args)
 
 
-def commit_changes(changes):
+def commit_changes(
+    changes: t.List[str],
+) -> t.Tuple[str, t.Optional[str], str, t.Optional[str]]:
```

Comment:
At the point where a function returns a four-tuple, one might consider returning a namedtuple or a frozen dataclass.

Reply:
Makes sense. Although I'd prefer to do this in a subsequent PR, since otherwise the change-set would be too big and harder to review.

Reply:
Indeed